internal/io/fs.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package io
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func CheckDirIsEmpty(target string) error {
+	if _, err := os.Stat(target); os.IsNotExist(err) {
+		return nil
+	}
+
+	return filepath.Walk(target, func(path string, info os.FileInfo, err error) error {
+		return fmt.Errorf("the target directory is not empty: %s", path)
```

Comment:
you might want to embed `err` here, to provide more information when the program exits